### ISSUE TITLE
Small amount of users experiencing kotlin.UninitializedPropertyAccessException crash

### ISSUE THREAD

**Issue by tarouboy** [RevenueCat/purchases-android#377]
- [x] I have updated Purchases SDK to the latest version
- [x] I have read the [Contribution Guidelines](https://github.com/RevenueCat/purchases-android/blob/main/CONTRIBUTING.md)
- [x] I have read the [Help Center](https://support.revenuecat.com/hc/en-us)
- [x] I have read [docs.revenuecat.com](https://docs.revenuecat.com/)
- [x] I have searched for [existing Github issues](https://github.com/RevenueCat/purchases-android/issues)

**Describe the bug**
There's a small amount of users (<1%) experiencing this kotlin.UninitializedPropertyAccessException crash.

1. Environment
   1. Platform: Android
   2. SDK version: 4.3.3
   3. OS version: All
   4. Android Studio version: 4.1
   5. How widespread is the issue. Percentage of devices affected. <1%
2. [Debug logs](https://docs.revenuecat.com/docs/debugging) that reproduce the issue

```
kotlin.UninitializedPropertyAccessException: 
  at com.revenuecat.purchases.Purchases$Companion.getSharedInstance (Purchases.kt:1760)
  at com.revenuecat.purchases.hybridcommon.CommonKt.logIn (common.kt:195)
  at com.revenuecat.purchases.react.RNPurchasesModule.logIn (RNPurchasesModule.java:168)
  at java.lang.reflect.Method.invoke (Native Method)
  at com.facebook.react.bridge.JavaMethodWrapper.invoke (JavaMethodWrapper.java:372)
  at com.facebook.react.bridge.JavaModuleWrapper.invoke (JavaModuleWrapper.java:188)
  at com.facebook.react.bridge.queue.NativeRunnable.run (Native Method)
  at android.os.Handler.handleCallback (Handler.java:938)
  at android.os.Handler.dispatchMessage (Handler.java:99)
  at com.facebook.react.bridge.queue.MessageQueueThreadHandler.dispatchMessage (MessageQueueThreadHandler.java:27)
  at android.os.Looper.loop (Looper.java:263)
  at com.facebook.react.bridge.queue.MessageQueueThreadImpl$4.run (MessageQueueThreadImpl.java:226)
  at java.lang.Thread.run (Thread.java:923)
```
  
3. Steps to reproduce, with a description of expected vs. actual behavior

Can't reproduce on my side.
Constantly receives from Google Play crash reports and Sentry reports.

4. **Other information** (e.g. stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, etc.)

I have no experience with kotlin but I believe there's someway to check if the sharedInstance is still there before continuing? Expect to throw an error instead of crashing.

**Additional context**
N/A

**Comment by tarouboy** [comment]
> Hello! It doesn't seem like we have quite enough information to send this to a human yet to help out. We would love if you could provide more details about your issue by following the template without modifying any of the pre-filled text. If you're looking for support, head over to our [Help Center](https://support.revenuecat.com/hc/en-us) to get in touch with our team directly.

Updated.

**Comment by vegaro** [comment]
Where are you configuring the SDK in your app? This crash occurs when trying to access any function before the SDK has been configured. Is it possible you have a race condition between `logIn` and where you're setting up the SDK?

**Comment by vegaro** [comment]
Where are you configuring the SDK in your app? This crash occurs when trying to access any function before the SDK has been configured. Is it possible you have a race condition between `logIn` and where you're setting up the SDK?

**Comment by tarouboy** [comment]
I understand and I configured the setup code in the first line already.

Is there a way to check if the SharedInstance exists before executing logIn?

Or is it possible to expose a function to check if the SharedInstance exists? so I will run the setup code again?

If both not possible, is it ok to run the setup code multiple times? So I will run the setup code every time right before logIn.

**Comment by vegaro** [comment]
Do you mind sharing a snippet of where you are calling setup and login? In our sample https://github.com/RevenueCat/react-native-purchases/blob/main/examples/MagicWeather/App.js#L30 we recommend calling setup it in App creation. 

We have been getting a bunch of reports of this, and we would like to understand why this is happening and what we can do to fix it. The solutions you propose should work, but are not the best way to solve this. This could happen with any function, not only login, so it would mean you would have to check if the shared instance exists before calling any function, which is not ideal. Calling setup multiple times could have unexpected results, so I wouldn't recommend it either.

**Comment by vegaro** [comment]
I have released 4.5.0 that adds `isConfigured`, it can be used to prevent the crash. We will add that to react native soon. We will also make sure the Exception is thrown as a JS error and not as a native exception that is impossible to catch.
